bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/handler/SamsungTvHandler.java

```diff
@@ -81,301 +106,632 @@ public class SamsungTvHandler extends BaseThingHandler implements RegistryListen
+                }
+                break;
+            case PROTOCOL_WEBSOCKET:
+            case PROTOCOL_SECUREWEBSOCKET:
+                initializeConfig();
+                if (!initialized) {
+                    logger.warn("{}: TV binding is not yet Initialized", host);
+                }
+                break;
+            case PROTOCOL_LEGACY:
+                initialized = true;
+                break;
+        }
+        showConfiguration();
+    }
+
+    public void initializeConfig() {
+        if (!initialized) {
+            TVProperties properties = fetchTVProperties(0, 0);
+            if ("on".equals(properties.getPowerState())) {
+                updateSettings(properties);
+            }
+        }
+    }
+
+    public void updateSettings(TVProperties properties) {
+        setPowerState("on".equals(properties.getPowerState()));
+        setModelName(properties.getModelName());
+        int year = Integer.parseInt(properties.getModel().substring(0, 2));
+        if (properties.getFrameTVSupport() && year >= 22) {
+            logger.warn("{}: Art Mode is NOT SUPPORTED on Frame TV's after 2021 model year", host);
+            setArtMode2022(true);
+            artApiVersion = 1;
+        }
+        setArtModeSupported(properties.getFrameTVSupport() && year < 22);
+        logger.debug("{}: Updated artModeSupported: {} PowerState: {}({}) artMode2022: {}", host, getArtModeSupported(),
+                getPowerState(), properties.getPowerState(), getArtMode2022());
+        initialized = true;
+    }
+
+    public void showConfiguration() {
+        logger.debug("{}: Configuration: {}, port: {}, token: {}, MAC: {}, subscription: {}", host,
+                configuration.getProtocol(), configuration.getPort(), configuration.getWebsocketToken(),
+                configuration.getMacAddress(), configuration.getSubscription());
+        if (configuration.isWebsocketProtocol()) {
+            if (configuration.getSmartThingsApiKey().isBlank()) {
+                logger.debug("{}: SmartThings disabled", host);
+            } else {
+                logger.debug("{}: SmartThings enabled, device id: {}", host, configuration.getSmartThingsDeviceId());
+            }
         }
+    }
+
+    /**
+     * get PowerState from TVProperties
+     * Note: Series 7 TV's do not have the PowerState value
+     *
+     * @return String giving power state (TV can be on or standby, off if unreachable)
+     */
+    public String fetchPowerState() {
+        logger.trace("{}: fetching TV Power State", host);
+        TVProperties properties = fetchTVProperties(0, 2);
+        String PowerState = properties.getPowerState();
+        setPowerState("on".equals(PowerState));
+        logger.debug("{}: PowerState is: {}", host, PowerState);
+        return PowerState;
+    }
+
+    public boolean handleCommand(String channel, Command command, int ms) {
+        logger.info("{}: Scheduling future command {}: {} in {}ms", host, channel, command, ms);
```

Comment:
I think this can be removed. or if needed lowered to trace

Reply:
Removed